telemetry_api/lib/telemetry_api/traces.ex

```diff
@@ -21,22 +21,148 @@ defmodule TelemetryApi.Traces do
       {:ok, "merkle_root"}
   """
   def create_task_trace(merkle_root) do
-    span_ctx = Tracer.start_span(
-      "Task: #{merkle_root}",
-      %{attributes: [
-        {:merkle_root, merkle_root}
-        ]}
+    span_ctx =
+      Tracer.start_span(
+        "Task: #{merkle_root}",
+        %{
+          attributes: [
+            {:merkle_root, merkle_root}
+          ]
+        }
       )
+
     ctx = Ctx.get_current()
-    
+
     TraceStore.store_trace(merkle_root, %Trace{
       parent_span: span_ctx,
       context: ctx
     })
-    IO.inspect(merkle_root)
+
+    IO.inspect("New task trace with merkle_root: #{IO.inspect(merkle_root)}")
     {:ok, merkle_root}
   end
 
+  @doc """
+  Registers an operator response in the task trace.
+
+  ## Examples
+
+      iex> merkle_root = "0x1234567890abcdef"
+      iex> operator_id = "0x..."
+      iex> register_operator_response(merkle_root, operator_id)
+      :ok
+  """
+  def register_operator_response(merkle_root, operator_id) do
+    case TraceStore.get_trace(merkle_root) do
+      nil ->
+        IO.inspect("Context not found for #{merkle_root}")
+        {:error, "Context not found for #{merkle_root}"}
+
+      trace ->
+        Ctx.attach(trace.context)
+        Tracer.set_current_span(trace.parent_span)
+
+        Tracer.add_event(
+          "Operator ID: #{operator_id}",
+          [
+            {:merkle_root, merkle_root},
+            {:operator_id, operator_id}
+          ]
+        )
+
+        ctx = Ctx.get_current()
+
+        TraceStore.store_trace(
```

Comment:
Is it needed update the context on every event?

Reply:
It's not, you are right! Updated!